aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -893,11 +966,25 @@ def route(
         cors: Optional[bool] = None,
         compress: bool = False,
         cache_control: Optional[str] = None,
+        middleware: Optional[List[Callable]] = None,
     ):
         def register_route(func: Callable):
             # Convert methods to tuple. It needs to be hashable as its part of the self._routes dict key
             methods = (method,) if isinstance(method, str) else tuple(method)
-            self._routes[(rule, methods, cors, compress, cache_control)] = func
+
+            tuple_key = (rule, methods, cors, compress, cache_control)
```

Comment:
```suggestion
            route_key = (rule, methods, cors, compress, cache_control)
```

a more descriptive name for its intent